File: 0609. Find Duplicate File in System/Solution.py
class Solution:
    def findDuplicate(self, paths: List[str]) -> List[List[str]]:
        hashmap = dict()
        
        # Try to parse the path
        for path in paths:
            p = path.split()
            for p_ in p[1:]:
                dir_file = (p[0]+"/"+p_).rstrip(")").split("(")
                print(dir_file)
                
                if dir_file[1] not in hashmap:
                    hashmap[dir_file[1]] = [dir_file[0]]
                else:
                    hashmap[dir_file[1]] += [dir_file[0]]
        
        print(hashmap)
        
        res = []
        
        for k,v in hashmap.items():
            if len(v) > 1:
                res.append(v)
                
        return res
        
        

File: 0609. Find Duplicate File in System/Solution.cpp
vector<string> split(string sData, string sDelimiter)
{
    char *pToken = NULL;

    uint uStringLength = strlen(sData.c_str());

    char aData[uStringLength + 1];
    vector<string> vTokens;
    vTokens.clear();

    memset(aData, 0x00, uStringLength + 1);
    memcpy(aData, sData.c_str(), uStringLength);

    pToken = strtok(aData, sDelimiter.c_str());
    while (pToken != NULL)
    {
        vTokens.push_back(pToken);
        pToken = strtok(NULL, sDelimiter.c_str());
    }

    if (pToken != NULL)
        delete pToken;

    return vTokens;
}

std::string& rtrim(std::string& str, const std::string& chars = "\t\n\v\f\r ")
{
    str.erase(str.find_last_not_of(chars) + 1);
    return str;
}
 

class Solution {
public:
    vector<vector<string>> findDuplicate(vector<string>& paths) {
        std::map<std::string, std::vector<std::string>> hashmap;
        
        std::vector<vector<string>> ret;
        
        for (auto &path : paths)
        {
            auto tokens = split(path, " ");
            for (int i=1;i<tokens.size();i++)
            {
                auto current = split((tokens[0]+"/"+tokens[i]), "(");
                hashmap[rtrim(current[1], ")")].push_back(current[0]);
            }
        }
        
        for (auto & [k,v] : hashmap)
        {
            if (v.size() > 1)
            {
                ret.push_back(v);
            }
        }
        
        return ret;
        
    }
};
